docs/source/developers/contributing.rst

```diff
@@ -304,3 +304,40 @@ to your branch, which they sometimes do to help move a pull request along.
 In addition, the GitHub PR "suggestion" feature can also add commits to
 your branch, so it is possible that your local copy of your branch is missing
 some additions.
+
+Guidance for specific features
+==============================
+
+From time to time the community has discussions on specific types of features
+and improvements that they expect to support.  This section outlines decisions
+that have been made in this regard.
+
+Endianess
++++++++++
+Arrow is primarily a little endian format there has been some effort to
+support big endian platforms as well. Based on a mailing list discussion,
+The requirements for a new platform are:
+
+1. A robut (non-flaky, returns results in a reasonable time) Continuous integration setup.
+2. Performance benchmarks in performance critical parts of the code to demonstrate no
+   regression.
+
+Furthermore for big-endianess support there are two levels that an implementation can support
+1.  Native endianness (all arrow communication happens with processes of the same endianness.
+2.  Cross platform support (implementations will do byte reordering when appropriate for IPC
+    and flight messages).
+
```

Comment:
On some platforms (e.g. C++), there are additional works (e.g. Parquet support in C++) 

Can we create the third level? Or do we put these works into the second level?

Reply:
I would actually place that in level #1.  I meant level #2 applies only to Arrow specific transport technologies (IPC format and flight)

Reply:
Clarified in doc.